ably/auth.go

```diff
@@ -42,21 +63,87 @@ func (c Capability) String() string {
 }
 
 type Token struct {
```

Comment:
This is a `TokenDetails` struct, not a `Token` struct (it is confusing to have a `Token` struct with a `Toke` property). See TokenDetails under http://docs.ably.io/client-lib-development-guide/features/#types.  Can we ensure consistency with other client libraries here?